android/src/main/java/com/SplashPad/export_plugin/ExportPlugin.java

```diff
@@ -102,53 +88,60 @@ public void onMethodCall(@NonNull MethodCall call, @NonNull Result result) {
   @Override
   public boolean onActivityResult(int requestCode, int resultCode, Intent intent) {
 
-    //GUARD
+    if (BuildConfig.DEBUG) {
+        Log.d("ExportPlugin", "onActivityResult: intent = " + intent);
+    }
+
+    //GUARD : only handle create file requests
     if (requestCode != CREATE_FILE) return false;
 
-    if (BuildConfig.DEBUG) {
-      Log.d("ExportPlugin", "onActivityResult: intent = " + intent);
+    if (resultCode == RESULT_CANCELED) {
+        // If we get a cancelled result, we should reset the result and path
+        result.error("result cancelled", null, null);
+        result = null;
+        path = null;
+        return false;
     }
 
-    try {
-      if (intent != null && intent.getData() != null) {
-        writeFile(intent.getData());
-        result.success("Successfully saved to local storage");
-      } else {
-        result.error(null, "No directory chosen", null);
-      }
-    } catch (final Throwable e) {
-      result.error(null,"Failed to save file", e);
-    } finally {
-      result = null;
-      path = null;
+    if (resultCode == RESULT_OK) {
+        // GUARD : to make sure this block does not run even after cancelled
+        // result is null only in two conditions, a success RESULT_OK return, or a RESULT_CANCELLED return
+        if (result == null) {
+            return false;
+        }
+
+        try {
+            if (intent != null && intent.getData() != null) {
+                writeFile(intent.getData());
+                result.success("Successfully saved to local storage");
+            } else {
+                result.error("No directory chosen", null, null);
+            }
+        } catch (final Throwable e) {
+            result.error("Failed to save file",null, e);
+        } finally {
+            result = null;
+            path = null;
+        }
+        return true;
     }
 
-    return true;
+    // Unhandled case
+    return false;
   }
 
-  private void launchFilePicker(String title) {
+  private void launchFilePicker(String title, String mimeType) {
     // file picker prompt to save file
     final Intent intent = new Intent();
     intent.setAction(Intent.ACTION_CREATE_DOCUMENT);
     intent.addCategory(Intent.CATEGORY_OPENABLE);
-    intent.setType("audio/wav");
+    // setType does not seem to affect outcome audio/video, but we are setting it anyway
+    intent.setType(mimeType);
```

Comment:
it is odd that we were correctly generating .mp4 files even though this is hardcoded to audio/wav. I have set it to use the proper mimeType anyway